contrib/opencensus-ext-azure/opencensus/ext/azure/common/transport.py

```diff
@@ -45,25 +48,44 @@ def _transmit(self, envelopes):
         if not envelopes:
             return 0
         try:
+            headers = {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json; charset=utf-8',
+            }
+            if self.options.credential:
+                token = self.options.credential.get_token(_MONITOR_OAUTH_SCOPE)
+                headers["Authorization"] = "Bearer {}".format(token.token)
             response = requests.post(
                 url=self.options.endpoint,
                 data=json.dumps(envelopes),
-                headers={
-                    'Accept': 'application/json',
-                    'Content-Type': 'application/json; charset=utf-8',
-                },
+                headers=headers,
                 timeout=self.options.timeout,
                 proxies=json.loads(self.options.proxies),
             )
         except requests.Timeout:
             logger.warning(
                 'Request time out. Ingestion may be backed up. Retrying.')
             return self.options.minimum_retry_interval
-        except Exception as ex:  # TODO: consider RequestException
+        except requests.RequestException as ex:
             logger.warning(
                 'Retrying due to transient client side error %s.', ex)
             # client side error (retryable)
             return self.options.minimum_retry_interval
+        except CredentialUnavailableError as ex:
+            logger.warning(
+                'Error with credential configuration %s', ex
```

Comment:
Not sure if the friendly message will always be in sync with the actual exception.  My suggestion is to simply say 'Credential error' and append the exception message.  What might be useful is to add 'dropping telemetry' coz that is something the ex message wont convey.

Reply:
Refer to the [error handling](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/identity/Azure.Identity/README.md#error-handling) section of azure identity
s README. `CredentialUnavailableError` is only returned if the configuration is incorrect. This is the contract that azure identity has given for their apis, and they must respect it. I think the message is an appropriate blanket statement for all of the errors that can throw this exception.

The reason why I did not add "Dropping telemetry" is because we still have not decided whether or not we are going to do this (as a team). As noted above in the PR description, the exception handling is still a WIP.

Reply:
At best you are duplicating the message you are going to log.  If you are going to do anything addition to what the exception message already says then dont add extra stuff.  All it will take it is for Azure SDK to throw CredentialUnavailableError for another scenario and your message will be out of sync.

Reply:
Alright, will change the error message.